tests/test_1d/test_backward_1d.py

```diff
@@ -161,3 +161,50 @@ def test_t2_backward_CPU_points(
     inputs = (points, targets)
 
     assert gradcheck(apply_finufft1d2(fftshift, isign), inputs, eps=1e-8, atol=1e-5 * N)
+
+
+def check_t2_backward(
+    N: int,
+    modifier: int,
+    fftshift: bool,
+    isign: int,
+    device: str,
+    points_or_targets: bool,
+) -> None:
+    points = torch.rand((1, N + modifier), dtype=torch.float64).to(device) * 2 * np.pi
+    targets = torch.randn(N, dtype=torch.complex128).to(device)
+
+    points.requires_grad = points_or_targets
+    targets.requires_grad = not points_or_targets
+
+    inputs = (points, targets)
+
+    def func(points, targets):
+        return pytorch_finufft.functional.finufft_type2.apply(
+            points,
+            targets,
+            None,
+            dict(modeord=int(not fftshift), isign=isign),
+        )
+
+    assert gradcheck(func, inputs, atol=1.5e-4 * N)
+
+
+@pytest.mark.parametrize("N", Ns)
+@pytest.mark.parametrize("modifier", length_modifiers)
+@pytest.mark.parametrize("fftshift", [False, True])
+@pytest.mark.parametrize("isign", [-1, 1])
+def test_t2_backward_CPU_values(
+    N: int, modifier: int, fftshift: bool, isign: int
+) -> None:
+    check_t2_backward(N, modifier, fftshift, isign, "cpu", False)
```

Comment:
If you want, the cuda tests should be obvious extensions here and CI should pick them up for you

Reply:
good point, added those